src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -0,0 +1,1104 @@
+                                              aPositionTag, aSurfaceTag);
+
+        // verify cluster requirements concerning valid fields and field relationships
+        if (!IsValidSupportedLocation(aNewLocation))
+        {
+            ChipLogError(Zcl, "ModifySupportedLocation %u - not a valid location object", aNewLocation.locationID);
+            return false;
+        }
+
+        // Updated location description must not match another existing location description.
+        // We ignore comparing the location ID as one of the locations will match this one.
+        if (!IsUniqueSupportedLocation(aNewLocation, true))
+        {
+            ChipLogError(Zcl, "ModifySupportedLocation %u - not a unique location object", aNewLocation.locationID);
+            return false;
+        }
+
+        // Replace the supported location with the modified location.
+        if (!mDelegate->ModifySupportedLocation(listIndex, aNewLocation))
+        {
+            return false;
+        }
+    }
+
+    if (mapIDChanged)
+    {
+        mDelegate->HandleSupportedLocationsUpdated();
+    }
+
+    NotifySupportedLocationsChanged();
+    return true;
+}
+
+bool Instance::ClearSupportedLocations()
+{
+    // does device mode allow this attribute to be updated?
+    if (!mDelegate->IsSupportedLocationsChangeAllowed())
+    {
+        return false;
+    }
+
+    if (mDelegate->ClearSupportedLocations())
+    {
+        mDelegate->HandleSupportedLocationsUpdated();
+        NotifySupportedLocationsChanged();
+        return true;
+    }
+
+    return false;
+}
+
+//*************************************************************************
+// Supported Maps manipulators
+
+bool Instance::IsSupportedMap(uint8_t aMapId)
+{
+    uint32_t dummyIndex;
+    MapStructureWrapper dummyMap;
+
+    return mDelegate->GetSupportedMapById(aMapId, dummyIndex, dummyMap);
+}
+
+bool Instance::AddSupportedMap(uint8_t aMapId, const CharSpan & aMapName)
+{
+    uint8_t mapIndex = 0;
+    MapStructureWrapper entry;
+
+    // check max# of list entries
+    if (mDelegate->GetNumberOfSupportedMaps() >= kMaxNumSupportedMaps)
+    {
+        ChipLogError(Zcl, "AddSupportedMap %u - maximum number of entries", aMapId);
+        return false;
+    }
+
+    //  Map name SHALL include readable text that describes the map name (cannot be empty string).
+    if (aMapName.empty())
+    {
+        ChipLogError(Zcl, "AddSupportedMap %u - Name must not be empty string", aMapId);
+        return false;
+    }
+
+    // Each entry in this list SHALL have a unique value for the Name field.
+    while (mDelegate->GetSupportedMapByIndex(mapIndex++, entry))
+    {
+        // the name cannot be the same as an existing map
+        if (entry.IsNameEqual(aMapName))
+        {
+            ChipLogError(Zcl, "AddSupportedMap %u - A map already exists with same name '%s'", aMapId, entry.GetName().data());
```

Comment:
This is using %s to print non-null-terminated buffers...  The right way to do this is:

```suggestion
            ChipLogError(Zcl, "AddSupportedMap %u - A map already exists with same name '%.*s'", aMapId, static_cast<int>(entry.GetName().size()), entry.GetName().data());
```

I did say that in https://github.com/project-chip/connectedhomeip/pull/33991#discussion_r1665196689, no ?

Reply:
I'm sure I used this somewhere else. I must have figured it out later. I have fixed this [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/57818b71d846aa8fdc841bcd28666d138ad1ede2).